hs_app_timeseries/migrations/custom_data_migration_20160718.py

```diff
@@ -0,0 +1,37 @@
+import logging
+
+from django.db import migrations
+
+from hs_core.hydroshare.utils import resource_modified
+from hs_app_timeseries.models import TimeSeriesResource
+
+
+def delete_extracted_metadata(apps, schema_editor):
+    # For all existing timeseries resources, delete all resource specific metadata.
+    # This way we can keep the resource GUID and allow the resource owners to manually
+    # delete the existing SQlite file and then add a new SQlite file to regenerate
+    # resource specific metadata
+    log = logging.getLogger()
+    for res in TimeSeriesResource.objects.all():
```

Comment:
Failing to do it this way adds an implicit dependency to the state of the model as it is in the codebase at the time this migration is run (which may not match what was in the codebase at the time this code was committed), as opposed to _what the migration machinery thinks this model should be when it runs this migration_.